client/components/googleMaps.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Loader } from '@googlemaps/js-api-loader';
+
+const loader = new Loader({
+  apiKey: process.env.GOOGLE_API,
+  version: 'weekly',
+  libraries: ['places', 'marker', 'drawing']
+});
+
+export default class GoogleMaps extends React.Component {
+  constructor(props) {
+    super(props);
+    this.googleMapDiv = React.createRef();
+    this.createMap = this.createMap.bind(this);
+  }
+
+  componentDidMount() {
+    this.createMap();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.results !== prevProps.results) {
+      this.createMap();
+    }
+  }
+
+  createMap() {
+    const states = this.props.results;
+    const mapOptions = {
+      center: { lat: 37.0902, lng: -95.7129 },
+      zoom: 4,
+      streetViewControl: false,
+      mapTypeControl: false
+    };
+    loader.load().then(google => {
+      const map = this.map = new window.google.maps.Map(this.googleMapDiv.current, mapOptions);
+      this.setState({
+        google,
+        map
+      });
+      return map;
+    })
+      .then(map => {
+        if (states.length !== 0) {
+          const bound = new window.google.maps.LatLngBounds();
+
+          states.map(state => {
+            const destination = {
+              lat: Number(state.latitude),
+              lng: Number(state.longitude)
+            };
+            if (state.latitude === '' || state.longititude === '') {
+              return null;
+            }
+            bound.extend(new window.google.maps.LatLng(destination));
+            const marker = new window.google.maps.Marker({
+              position: destination,
+              map
+            });
+            const infoWindow = new window.google.maps.InfoWindow({
+              content:
+                '<h6 class="info-title">' + state.name + '</h6>' +
+                '<p class="info-window mb-0">' + state.addresses[0].line1 + '</p>' +
+                '<p class="info-window gold fw-bold">' + state.addresses[0].city + ', ' + state.addresses[0].stateCode + ' ' + state.addresses[0].postalCode + '</p>',
```

Comment:
To be safe and avoid Cross-Site Scripting attacks, it's important to escape the strings you bring in from untrusted sources (strings you didn't add yourself, as the developer). Here is a function you can use.

https://www.npmjs.com/package/escape-html

More info about XSS can be found here:
https://owasp.org/www-community/attacks/xss/

I did some trawling online, and it looks like the google maps sdk devs themselves use DOM creation instead of HTML concatenation. 

https://github.com/googlemaps/js-samples/pull/773/files

I tried finding some other discussions but most of the people involved seemed to dismiss the notion entirely (because they're irresponsible i guess?)